internal/expr/expr_test.go

```diff
@@ -73,6 +73,20 @@ var tests = []struct {
 		foo = bar
 		and
 		x = y]]`,
+}, {
+	"comments",
+	`SELECT &Person.* -- The line with &Person.* on it
+	 FROM person /* The start of a multi line comment
+	 It keeps going here with some weird chars /-*"*/
+	 And now it stops */ WHERE x = /-*'' -- The WHERE line
+	 AND y = $Person.id -- The line with $Person.id on it
+	 `,
+	`[Bypass[SELECT ] Output[[] [Person.*]] Bypass[ -- The line with &Person.* on it
+	 FROM person /* The start of a multi line comment
+	 It keeps going here with some weird chars /-*"*/
+	 And now it stops */ WHERE x = /-*'' -- The WHERE line
+	 AND y = ] Input[Person.id] Bypass[ -- The line with $Person.id on it
+	 ]]`,
```

Comment:
I'm not sure whether the addition to `advance` handles this but either way, is it worth also testing `SELECT (&Person.foo, /* ... */ &Person.bar) ...` and `SELECT (&Person.foo /* ... */, &Person.bar) ...` and the same for single-line comments?

Reply:
These actually would break the parser. To fix this it would probably require some pretty big changes. The easiest way to do that would be to do an initial pass of the input string to remove comments. I'll have a chat with Gustavo about the approach to take here since if we kept this code as it is it would probably be good enough in 99% of use cases.

Reply:
Actually I take that back, the examples above would work, but this wouldn't `* AS (&Person.foo, /* ...*/, &Person.bar)`. This is because we would parse those as separate expressions but not this one.